File: src/main/java/com/wikia/tibia/repositories/LocationRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.LocationGateway;
import com.wikia.tibia.objects.Location;

public class LocationRepository extends WikiObjectRepository {

    public LocationRepository() {
        super(Location.class, new LocationGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/LocationRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Location

class LocationRepository : WikiObjectRepository<Location>(
        wikiObjectClass = Location::class.java,
        wikiObjectGateway = WikiObjectGateway(Contract.LOCATIONS)
)